packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -559,19 +577,30 @@ const convertInputToLatestMessageText = (
     return input;
   }
 
-  const lastUserMessageString = input.findLast(
-    (message): message is UserMessage =>
-      (message.type === "message" || !message.type) &&
-      message.role === "user" &&
-      !!message.content
-  )?.content;
-
-  if (!lastUserMessageString) {
+  const lastUserMessage = input.findLast(isUserMessage);
+  if (!lastUserMessage) {
     throw makeBadRequestError({
       error: new Error("No user message found in input"),
       headers,
     });
   }
 
-  return lastUserMessageString;
+  return formatUserMessageContent(lastUserMessage.content);
+};
+
+type UserMessage = {
+  role: "user";
+  content: UserMessageContent;
+};
+
+const isUserMessage = (message: any): message is UserMessage =>
+  (message.type === "message" || !message.type) &&
+  message.role === "user" &&
+  !!message.content;
```

Comment:
can you use zod to help? maybe can take subtype from the input schema

and i think it'd be preferrable to use `message: unknown` rather than  `any`

Reply:
yeah I'll take a look at using zod to apply the type!